### ISSUE TITLE
silence warnings in the terminal

### ISSUE THREAD

**Issue by xxchan** [rust-lang/cargo#14258]
I want to silence warnings without re-compiling, so I don't want to use `RUSTFLAGS="-A warnings"`.

---

I'm here cause I was wanting a similar thing... I wish there was a way to run something like 
`cargo run --silence-warnings` or `cargo build --silence-warnings` that silences all warnings except fatal errors. Maybe might print a single line telling you that warnings exist without the visual clutter of the warnings. It should still allow output logs from the compiled binary.

The main reason is that I work on some massive code bases and there's dead code and unused imports, etc everywhere on the dev branches and when I need to only focus on either actual fatal Errors. Sometimes there's so many warnings that it exceeds my terminal's max history and I can't even read the actual Errors that are thrown that my terminal that are preventing my code from compiling or running. Having to go around and put #![allow(foo_bar)] everywhere and then remove it would just take too much time.But that's currently one of my only options. Please, please, please do something about this. Not having this feature is one of the current banes of my existence :D

_Originally posted by @noraa-july-stoke in https://github.com/rust-lang/cargo/issues/3591#issuecomment-1703234511_
         
--- 
   
Related #8424

**Comment by rasheedmhd** [comment]
Here for the same reason

**Comment by epage** [comment]
FYI our current plan is for our solution to #8424 to also solve this, see #14388

**Comment by epage** [comment]
Tracking issue: #14802

**Comment by bernborgess** [comment]
TLDR: For complete _silence_ use
```rust
RUSTFLAGS=-Awarnings cargo run -q  
```

**Comment by weihanglo** [comment]
@bernborgess it does not always work for some target platforms until 1.86. See <https://github.com/rust-lang/cargo/pull/15036>.

I would recommend people taking a look at the unstable feature [`build.warnings`](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#buildwarnings), and, if willing, provide feedback :)
